src/lib/hooks/useLoadResizeHook.js

```diff
@@ -0,0 +1,34 @@
+import { useLayoutEffect } from 'react';
+import { getElementsPositionDifference } from '../services/elementPositionService';
+
+export const useLoadResize = (effect, dependencies, contentEl, viewportEl, wait) => {
+  let throttleTimeout = null;
+
+  const callBack = () => {
+    effect(getElementsPositionDifference(contentEl, viewportEl));
+    throttleTimeout = null;
+  };
+
+  useLayoutEffect(() => {
+    const handleLoadResize = () => {
+      if (wait) {
+        if (throttleTimeout === null) {
+          // eslint-disable-next-line react-hooks/exhaustive-deps
+          throttleTimeout = setTimeout(callBack, wait);
+        }
+      } else {
+        callBack();
+      }
+    };
+
+    window.addEventListener('load', handleLoadResize);
+    window.addEventListener('resize', handleLoadResize);
+
+    return () => {
+      window.removeEventListener('load', handleLoadResize);
```

Comment:
`clearTimeout` need to be added here in case we unmount component earlier than setTimeout callback is called